cmd/internal/applicationsnapshot/validation.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright © 2022 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package applicationsnapshot
+
+import (
+	_ "embed"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/xeipuuv/gojsonschema"
+	crd "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+)
+
+// Validate the jsonstring with the application snapshot schema.
+func (s *applicationSnapshot) Validate(jsonstring string) error {
+	// Struct to unmarshal the OpenAPIV3Schema required for validation.
+	var schemaProps struct {
+		Properties struct {
+			Spec *crd.JSONSchemaProps `json:"spec"`
+		} `json:"properties"`
+	}
+
+	//TODO: there must be a better way to avoid those 3 steps
+	// Marshal the openapi schema to JSON
+	openapiV3Schema, err := json.Marshal(s.Spec.Versions[0].Schema.OpenAPIV3Schema)
```

Comment:
We need to be careful about which version we use here, version at index 0 could be the one that is not served any more, that is I think we should filter on `CustomResourceDefinitionVersion.Served == true`

Reply:
Correct, changed it.